### PR TITLE
[MM-15297] Migrate "Preference.Get" to Sync by default #10721

### LINKED ISSUE #10721: [MM-15297] Migrate "Preference.Get" to Sync by default
If you're interested please comment here and come [join our "Contributors" community channel](https://community.mattermost.com/core/channels/tickets) on our daily build server, where you can discuss questions with community members and the Mattermost core team. For technical advice or questions, please  [join our "Developers" community channel](https://community.mattermost.com/core/channels/developers).

New contributors please see our [Developer's Guide](https://developers.mattermost.com/contribute/getting-started/).

----

**Notes**: [Jira ticket](https://mattermost.atlassian.net/browse/MM-15297)

Mattermost is migrating its Store layer to be sync by default, and only use Async when needed and we're looking for contributors to help with that effort. This Help Wanted issue is to migrate the `Get` in the `Preference` store.

The expected way to implement it is, go to the `Preference` store implementation in the `store/sqlstore/` directory, modify the method `Get` to return directly an object from the `model` module, and a `*model.AppError` (removing the `store.Do` wrapper). After that, you must modify the interface defined in `store/store.go` to match with the changes. Then, you should execute `make store-mocks` to rebuild the mocks with the new interface. And finally, modify the rest of the code (tests included) to use the new interface of the function properly.

Example: mattermost/mattermost-server#10613

### PR DESCRIPTION
**Pull request by ivanaairenee**
**Summary**
Migrate "Preference.Get" to Sync by default

[**Ticket Link**](https://mattermost.atlassian.net/browse/MM-15297)

Fixes #10721

### CODE REVIEW

**Comment by jespino** [comment]
@ivanaairenee Hi, let me know if you need any help with this PR

File: app/email_batching.go [hunk 1]
```diff
@@ -171,12 +171,12 @@ func (job *EmailBatchingJob) checkPendingNotifications(now time.Time, handler fu
 
 		// get how long we need to wait to send notifications to the user
 		var interval int64
-		pchan := job.server.Store.Preference().Get(userId, model.PREFERENCE_CATEGORY_NOTIFICATIONS, model.PREFERENCE_NAME_EMAIL_INTERVAL)
-		if result := <-pchan; result.Err != nil {
+		data, err := job.server.Store.Preference().Get(userId, model.PREFERENCE_CATEGORY_NOTIFICATIONS, model.PREFERENCE_NAME_EMAIL_INTERVAL)
```
Comment (jespino) [inline app/email_batching.go]: You can create the `preference` variable directly here, instead of `data` and then `preference := data`

Comment (ivanaairenee) [reply app/email_batching.go]: thanks, updated

File: app/preference.go [hunk 2]
```diff
@@ -32,13 +32,7 @@ func (a *App) GetPreferenceByCategoryForUser(userId string, category string) (mo
...
 
 func (a *App) GetPreferenceByCategoryAndNameForUser(userId string, category string, preferenceName string) (*model.Preference, *model.AppError) {
-	result := <-a.Srv.Store.Preference().Get(userId, category, preferenceName)
-	if result.Err != nil {
-		result.Err.StatusCode = http.StatusBadRequest
-		return nil, result.Err
-	}
-	data := result.Data.(model.Preference)
-	return &data, nil
+	return a.Srv.Store.Preference().Get(userId, category, preferenceName)
```
Comment (jespino) [inline app/preference.go]: This is not exactly the same behavior than before, the previous version was returning `http.StatusBadRequest` for the error here.

**Pull request merged**